arch/LoongArch/LoongArchInstPrinter.c

```diff
@@ -53,9 +53,13 @@
 static void printInst(MCInst *MI, uint64_t Address, const char *Annot,
 		      SStream *O)
 {
+	if (!map_use_alias_details(MI)) {
+		SStream_Close(O);
+		printInstruction(MI, Address, O);
+		SStream_Open(O);
+	}
 	if (!printAliasInstr(MI, Address, O))
 		printInstruction(MI, Address, O);
-	;
```

Comment:
This logic seems off. If `map_use_alias_details` is false, it calls `printInstruction` two times. Which is prone to introduce bugs IMHO.

Maybe change to:
```c
bool useAliasDetails = map_use_alias_details(MI);
map_set_fill_detail_ops(MI, useAliasDetails);

bool isAlias = printAliasInstr(MI, Address, O);

MCInst_setIsAlias(MI, isAlias);
if (!isAlias || !useAliasDetails) {
	map_set_fill_detail_ops(MI, !(isAlias && useAliasDetails));
	if (isAlias)
		SStream_Close(O);
	printInstruction(MI, Address, O);
	if (isAlias)
		SStream_Open(O);
}
```

It requires that you check in the operand handlers for `map_fill_detail_ops()` and return early.

Reply:
This approach seems more elegant.